arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/TerminalOps.kt

```diff
@@ -16,6 +16,22 @@ package arrow.fx.coroutines.stream
   suspend fun drain(): Unit =
     foldChunks(Unit) { _, _ -> Unit }
 
+  /**
+   * Similar to [drain] but applying an effect on each of the objects emitted.
+   * 
+   * This equivalent to doing:
+   * 
+   * ```
+   * stream.effectTap { **operation** }
+   *   .compile()
+   *   .drain()
+   * ```
+   * 
+   * The main difference, apart from being shorter, is that the operation is applied after we compile the stream into [TerminalOps] 
+   */
+  suspend fun forEach(block: (O) -> Unit): Unit =
```

Comment:
😱 `forEach`

I think we should implement `fold` here as an overload of `foldChunks` instead of `forEach`, which is in line with the APIs and naming we use in Arrow Core.